### PR TITLE
doc: Fix typos and grammar in ecbackend.rst

### PR DESCRIPTION
**Pull request by PracticalMetal**
Signed-off-by: Pulkit Mittal <2pulkit2@gmail.com> as part of OPEN SOURCE DAY

Fixed several typos and grammar in ceph/doc/dev/osd_internals/erasure_coding/ecbackend.rst

## Checklist
- [ ] References tracker ticket
- [x] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by varshar16** [comment]
Hi Pulkit, please see the [submitting patches doc ](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst)and update the patch with git commit --amend.

File: doc/dev/osd_internals/erasure_coding/ecbackend.rst [hunk 1]
```diff
@@ -2,18 +2,18 @@
 ECBackend Implementation Strategy
 =================================
 
-Misc initial design notes
+Miscellaneous initial design notes
 =========================
```
Comment (ideepika) [inline doc/dev/osd_internals/erasure_coding/ecbackend.rst:6]:
```suggestion
Miscellaneous initial design notes
==================================
```
the `===` is used for underlining in markdown, can you extend it to cover above text

File: doc/dev/osd_internals/erasure_coding/ecbackend.rst [hunk 2]
```diff
@@ -49,22 +49,22 @@ Distinguished acting set positions
 ----------------------------------
 
 With the replicated strategy, all replicas of a PG are
-interchangeable.  With erasure coding, different positions in the
-acting set have different pieces of the erasure coding scheme and are
+interchangeable.  With erasure-coding, different positions in the
```
Comment (ideepika) [inline doc/dev/osd_internals/erasure_coding/ecbackend.rst:52]: want to remove extra space before With?
`interchangeable. With`

File: doc/dev/osd_internals/erasure_coding/ecbackend.rst [hunk 3]
```diff
@@ -49,22 +49,22 @@ Distinguished acting set positions
...
+acting set have different pieces of the erasure-coding scheme and are
 not interchangeable.  Worse, crush might cause chunk 2 to be written
 to an OSD which happens already to contain an (old) copy of chunk 4.
 This means that the OSD and PG messages need to work in terms of a
-type like pair<shard_t, pg_t> in order to distinguish different pg
+type like a pair<shard_t, pg_t> in order to distinguish different pg
```
Comment (ideepika) [inline doc/dev/osd_internals/erasure_coding/ecbackend.rst:57]: here pair<> is a data structure type, don't think we need `a` for it

**Comment by ideepika** [comment]
@PracticalMetal  no need to close PR, you can reopen this one, this looks good overall, once you have fixed all the comments on your local branch, you can do: 

1. `git add -p` to add relevant changes addressing comments
2. use `git commit --amend` and read about it :) 
3. add signed-off line at the end of commit message when editor opens for amending commit message
4. don't forget, if it's vi/vim you need :q to exit it

**Comment by PracticalMetal** [comment]
I had some issue rebasing in commit and I will be submitting another PR with corrections, is it okay if I do so? @ideepika

**Comment by ideepika** [comment]
> I had some issue rebasing in commit and I will be submitting another PR with corrections, is it okay if I do so? @ideepika

that's fine!

File: doc/dev/osd_internals/erasure_coding/ecbackend.rst [hunk 4]
```diff
@@ -28,9 +28,9 @@ operation represented by the operation
 PGTemp and Crush
 ----------------
 
-Primaries are able to request a temp acting set mapping in order to
+Primaries can request a temp acting set mapping in order to
 allow an up-to-date OSD to serve requests while a new primary is
-backfilled (and for other reasons).  An erasure coded pg needs to be
+backfilled (and for other reasons).  An erasure-coded pg needs to be
```
Comment (ideepika) [inline doc/dev/osd_internals/erasure_coding/ecbackend.rst:33]: @PracticalMetal smallcaps pg can also be changed to capitals here

**Comment by PracticalMetal** [comment]
#42365 I've created a new PR please let me know if there is anything I can correct in that @ideepika

**Pull request closed without merging**